Extensions/Lighting/lightruntimeobject.js

```diff
@@ -0,0 +1,93 @@
+gdjs.LightRuntimeObject = function (runtimeScene, lightObjectData) {
+  gdjs.RuntimeObject.call(this, runtimeScene, lightObjectData);
+
+  this._radius =
+    lightObjectData.content.radius > 0 ? lightObjectData.content.radius : 1;
+  this._color = gdjs.LightRuntimeObject.hexToRGBColor(
+    lightObjectData.content.color
+  );
+  this._debugMode = lightObjectData.content.debugMode;
+  this._texture =
+    lightObjectData.content.texture === ''
+      ? null
+      : runtimeScene
+          .getGame()
+          .getImageManager()
+          .getPIXITexture(lightObjectData.content.texture);
+
+  if (gdjs.LightObstaclesManager)
+    this._obstacleManager = gdjs.LightObstaclesManager.getManager(runtimeScene);
```

Comment:
Don't declare your properties conditionnaly. Always declare them. It's the *value* that can change:

`this._obstaclesManager = condition ? something : null`

In other words, avoid things that are implicitly undefined at all cost.

Reply:
Also, nitpicking: `obstaclesManager`, to be consistent with the class name.